fix: remove start event initiator moved to or created within subprocess

smbea opened the pull request:
Closes #74

barmac commented on lib/RemoveInitiatorBehaviour.js:
```diff
@@ -0,0 +1,43 @@
+
+    var shape = context.shape;
+
+    if (is(shape, 'bpmn:StartEvent')) {
+
+      // if subProcess becomes the new parent
+      if (!is(context.oldParent, 'bpmn:SubProcess') && (is(context.newParent, 'bpmn:SubProcess'))) {
+
+        // remove initiator property
+        modeling.updateProperties(shape, { 'camunda:initiator': undefined });
+
+      }
+    }
```
Let's be explicit here. The conditions for the hook are:
* The moved/created shape is a Start Event.
* The start event has `camunda:initiator` (even if it's an empty string).
* New parent is a SubProcess.

What we don't care about is the old parent. The diagram could have been broken but behavior still needs to kick in.
Also, in the current implementation we trigger an unnecessary update if the start event does not have `camunda:initiator`.

My suggestion: check all three conditions and use `modeling.updateProperties` only if all of them are met.

barmac replied:
So the test case which is missing is when you move a start event which has the property and is already a child of a subprocess.

barmac reviewed (changes requested):
I've left a comment which needs to be addressed before we merge this PR. I like the test cases; the code is clean and well structured. Great job overall :)

marstamm commented:
I noticed that the Issue is linked in the PR description, but not in the commits. This makes it harder to find the corresponding issue once the PR is merged. Please consider adding it to one of your commits as well (cf. [Definition of Done](https://github.com/bpmn-io/internal-docs/blob/master/development/DEFINITION_OF_DONE.md))

smbea commented:
Changes overview:
- Added three conditions for property removal
- Added test case for when shape with `initiator` property is moved within subprocess
- Removed some `console.log`'s I accidentally left behind

- [On commits] Started adding issue link in commits' descriptions

barmac commented:
Congratulations on your first contribution 🎉

nikku commented:
:tada:
